htdocs/core/lib/functions.lib.php

```diff
@@ -2220,8 +2220,8 @@ function dol_escape_htmltag($stringtoescape, $keepb = 0, $keepn = 0, $noescapeta
 				} while ($diff);
 			}
 
-			$tmp = str_ireplace('&quot', '__DOUBLEQUOTE', $tmp);
-			$tmp = str_ireplace('&lt', '__LESSTAN', $tmp);
+			$tmp = str_ireplace('&quot;', '__DOUBLEQUOTE', $tmp);
```

Comment:
We don't want the semicolon because in hacking world the &xxx is often used without the semicolin. It works same way often and it bypass sanitization.

Reply:
Ok, the issue is then that:
`"` is replaced with __DOUBLEQUOTE
and
`&quot` is also replaced with __DOUBLEQUOTE which then becomes `";` .

So this means that we either need two different names for DOUBLEQUOTE or we need to convert `"` to "__DOUBLEQUOTE;" so that it becomes `&quot;` in the end.

I would think we cant the first approach, unless the `"` tokens need to be escaped which you did not implement (so therefore I think it's the first option (same as keeping " or &quot; in the end according to what was in the original.

Reply:
i see. I pushed a fix for this point into 2483c4ec58bc452d747796d41269b9a379d4fb0f